opencog/util/Logger.cc

```diff
@@ -461,6 +530,11 @@ void Logger::log(Logger::Level level, const std::string &txt)
     if (level <= backTraceLevel) flush();
 
     if (syncEnabled) flush();
+
+    while (logger_message_count != 0)
+    {
+        std::this_thread::yield();
+    }
```

Comment:
OK, I finally get the idea. But I don't think this is acceptable as it essentially makes `Logger::log` a blocking call, which we don't want to efficiency reason. And, as Linas said, we already have the `sync` flag for that.